### ISSUE TITLE
Versions of plugins from settings.gradle are not checked

### ISSUE THREAD

**Issue by koral--** [ben-manes/gradle-versions-plugin#367]
Steps to reproduce:
- use Gradle 6.x
- add some plugin with old version to `settings.gradle(.kts)` for example "com.gradle.enterprise"
- invoke `updateDependencies` task

```
plugins {
    id("com.gradle.enterprise") version "3.1"
}
```
At the time of writing there is a version 3.1.1.

More info about build scan plugin: https://docs.gradle.com/enterprise/gradle-plugin/#gradle_6_x_and_later

**Comment by ben-manes** [comment]
I suppose we would need to use [Settings](https://docs.gradle.org/current/javadoc/org/gradle/api/initialization/Settings.html) and evaluate both `getBuildscript()` and `getPluginManagement()`.

**Comment by nedtwigg** [comment]
Closely related, it also does not check for these in the `settings.gradle` file:

```gradle
pluginManagement {
  plugins { ...
```
```

**Comment by ben-manes** [comment]
Yes, see my comment above @nedtwigg. We would need to use both of those methods to capture those dependencies. Since those are not configurations, we'd have to make a detached configuration with the dependencies, repositories, and resolution strategy that those methods provide. Then resolve and print in the report with everything else.

**Comment by jaredsburrows** [comment]
It looks like `dependabot` is able to do this: https://github.com/jaredsburrows/cs-interview-questions/pull/29.

**Comment by ben-manes** [comment]
Yes, if I recall correctly they use simple regex parsing of the build files rather than executing it. That has other limitations.

The long-term approach that Github/Gradle are pursuing is outlined in this [proposal](https://docs.google.com/document/d/1TjxJJwgPavw-TFzK3110iH-CWstgdcVdb2JYiRy2GVs/edit) and [github issue](https://github.com/dependabot/dependabot-core/issues/1164). This would be similar to having static analyzers produce SARIF reports that the Github Security dashboard consumes, so it could simply be an action that feeds the current dependency graph into Github for a monitoring dashboard.

**Comment by Vampire** [comment]
https://github.com/jmfayard/refreshVersions can also handle it, so would be nice to have it here too :-)

**Comment by ben-manes** [comment]
PRs welcome. 😁

**Comment by Vampire** [comment]
As always and everywhere :-D

**Comment by ianbrandt** [comment]
Just adding a use case, the [Foojay Toolchains Plugin](https://github.com/gradle/foojay-toolchains#usage).

**Comment by ben-manes** [comment]
if someone wants to try this [idea](https://github.com/ben-manes/gradle-versions-plugin/issues/367#issuecomment-569784031) and send a pr that would be appreciated.

**Comment by Vampire** [comment]
Btw. imo `pluginManagement { plugins { ... } }` does not necessarily need to be checked. It's the same as plugins that are defined in a version catalog but not used anywhere. That block just defines versions that are used when a plugin later is applied without version. So if it is applied and thus used, it will already be covered (for project plugins already now, for settings plugins with this issue fixed).
